src/django_otp/forms.py

```diff
@@ -134,6 +139,19 @@ def _chosen_device(self, user):
         return device
 
     def _handle_challenge(self, device):
+        try:
+            generate_is_allowed, extra = device.generate_is_allowed()
+        except AttributeError:
+            raise forms.ValidationError(
```

Comment:
This error seems like a mismatch. But it can just be removed if `generate_is_allowed` is moved to the root class.

Reply:
As noted above, `generate_is_allowed` applies only to devices that generate token (`SideChannelDevice`). Therefore, if somehow the `generate_is_allowed` is called on a device that does not have it we notify the user.